src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Simplification/Simplifiers/CastSimplifier.cs

```diff
@@ -133,17 +106,7 @@ public static bool IsUnnecessaryAsCast(BinaryExpressionSyntax cast, SemanticMode
             if (expressionToCastType.IsExplicit)
             {
                 // Explicit reference conversions can cause an exception or data loss, hence can never be removed.
-                if (expressionToCastType.IsReference)
-                    return false;
-
-                // Unboxing conversions can cause a null ref exception, hence can never be removed.
-                if (expressionToCastType.IsUnboxing)
-                    return false;
-
-                // Don't remove any explicit numeric casts.
-                // https://github.com/dotnet/roslyn/issues/2987 tracks improving on this conservative approach.
-                if (expressionToCastType.IsNumeric)
-                    return false;
+                return false;
```

Comment:
i'm bulk disallowing these.  note that this breaks no existing tested scenarios, it just makes us us conservative and safer.  If we feel like there's ever an explicit cast which is safe to remove we can explicitly whitelist it. but we should start by bulk-disallowing this and not by having a black list (since any cases we miss is a bug that breaks user code.).